src/apps/replication/lib/mutation_log.cpp

```diff
@@ -564,7 +565,8 @@ void mutation_log::internal_write_callback(
 
     if (err == ERR_OK || err == ERR_HANDLE_EOF)
     {
-        dassert(g_end_offset == end_offset,
+        // the log may still be written when used for learning
```

Comment:
this is inevitable when the local app does not support async checkpointing, and it is already supported. Anyway, for apps like rocksdb, this won't happen as the checkpointing is asynchronous and this won't happen.